sdk/core/azure-core/test/ut/curl_connection_pool_test.cpp

```diff
@@ -617,17 +636,38 @@ namespace Azure { namespace Core { namespace Test {
       auto connection
           = CurlConnectionPool::g_curlConnectionPool.ExtractOrCreateCurlConnection(req, options);
 
+      auto failedCounter = 0;
+      for (auto i = 0; i < _detail::AzureSdkHttpbinRetryCount; i++)
       {
-        // Check we can use the connection to retrieve headers in the response, but the connection
-        // is still flagged as shutdown.
-        auto session
-            = std::make_unique<Azure::Core::Http::CurlSession>(req, std::move(connection), options);
+        GTEST_LOG_(INFO) << "forceConnectionClosed test iteration " << i << ".";
+        try
+        {
+          // Check we can use the connection to retrieve headers in the response, but the connection
+          // is still flagged as shutdown.
+          auto session = std::make_unique<Azure::Core::Http::CurlSession>(
+              req, std::move(connection), options);
+
+          auto r = session->Perform(Azure::Core::Context::ApplicationContext);
+          EXPECT_EQ(CURLE_OK, r);
+          auto response = session->ExtractResponse();
+          EXPECT_EQ(
+              response->GetStatusCode(), Azure::Core::Http::HttpStatusCode::SwitchingProtocols);
+          EXPECT_EQ("close", response->GetHeaders().find("Connection")->second);
+          break; // Test passed, no need to retry.
+        }
+        catch (Azure::Core::Http::TransportException const& ex)
+        {
 
-        auto r = session->Perform(Azure::Core::Context::ApplicationContext);
-        EXPECT_EQ(CURLE_OK, r);
-        auto response = session->ExtractResponse();
-        EXPECT_EQ(response->GetStatusCode(), Azure::Core::Http::HttpStatusCode::SwitchingProtocols);
-        EXPECT_EQ("close", response->GetHeaders().find("Connection")->second);
+          // CURL returns a connection error which triggers a transport exception.
```

Comment:
THis applies to all similar locations.

Reply:
> Is this behavior CURL specific?

From what I have seen so far, the errors occur on the `CurlConnection` ctor call, so they are curl specific, yes.

> If it is, this should be in a CURL specific conditional.

I am not sure what you mean by that. The test retries with this comment are in tests under `/ut/curl_*`, so they are already CURL specific. Those files are only conditionally included.
https://github.com/Azure/azure-sdk-for-cpp/blob/459c148d06c6c04645f5f8093245c1c285c3f7a4/sdk/core/azure-core/test/ut/CMakeLists.txt#L30-L34

Reply:
Sorry - I didn't notice that this was a curl specific test. But the exact same text appears in non-curl tests.

And my core point here is that this behavior (throwing a TransportException) *ISN'T* curl specific. It's true of ALL of our HTTP transports, and in general, anything that talks to a network has to be resiliant against network failures.

Since the behavior of throwing a transport exception on HTTP failures isn't specific to curl, I don't think that we should document that the behavior of throwing a transport exception is a curl specific behavior.

Reply:
> But the exact same text appears in non-curl tests.

Just to confirm we are seeing the same thing... **Where?**

> And my core point here is that this behavior (throwing a TransportException) ISN'T curl specific. It's true of ALL of our HTTP transports, and in general, anything that talks to a network has to be resiliant against network failures.

> Since the behavior of throwing a transport exception on HTTP failures isn't specific to curl, I don't think that we should document that the behavior of throwing a transport exception is a curl specific behavior.

That sounds good, will remove the comment, regardless.

Reply:
I left a comment at the other location I saw in transport_policy_options.cpp.